### PR TITLE
Improve E0585 help

### PR DESCRIPTION
**Pull request by Rageking8**

### CODE REVIEW

File: compiler/rustc_parse/src/parser/diagnostics.rs [hunk 1]
```diff
@@ -71,8 +71,8 @@ impl Error {
                     "found a documentation comment that doesn't document anything",
                 );
                 err.help(
-                    "doc comments must come before what they document, maybe a comment was \
-                          intended with `//`?",
+                    "doc comments must come before what they document, if a comment was \
+                          intended use `//`",
```
Comment (Rageking8) [inline compiler/rustc_parse/src/parser/diagnostics.rs:75]: I personally think the new phrasing is slightly more terse (slightly shorter for an already long line of output) and removes the questioning tone of the help message. However, if the change does not add any new value, do let me know, then I can close this PR. Thanks.

**Review by wesleywiser — APPROVED** [review]
This seems reasonable to me!

**Comment by wesleywiser** [comment]
@Rageking8 if you can rebase and resolve the merge conflict, I'd be happy to r+

**Comment by Rageking8** [comment]
Oops, new to git and cargo submodules changes keep slipping in. I think I fixed it.

**Comment by Rageking8** [comment]
@wesleywiser Fixed the merge conflicts and CI failures, may I get a rereview (added a few more changes to make CI pass). Thanks.

**Pull request merged**